doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,350 @@
+
+.. tabs::
+
+   .. group-tab:: Kconfig / prj.conf
+
+      .. code-block:: console
+
+         CONFIG_SB_SIGNING_KEY_FILE="priv.pem"
+
+   .. group-tab:: Command line
+
+      .. code-block:: console
+
+         -DCONFIG_SB_SIGNING_KEY_FILE=\"priv.pem\"
+
+      Escaped quotations avoid malformed-string warnings from Kconfig.
+
+This option only accepts the private key of an ECDSA key pair, as the public key will be extracted from it automatically during build time.
+
+The file argument must be a string and specified in one of the following ways:
+
+* The relative path to the file from the application directory.
+
+  * If the ``prj.conf`` file is external to the directory, the key's location will still be taken relative to the application directory, not to the configuration file.
+
+* The absolute path to the file.
+
+For example, if a directory named ``_keys`` and located in ``/home/user/ncs`` contains signing keys, then you can provide the path in the following ways:
+
+.. tabs::
+
+   .. group-tab:: Kconfig / prj.conf
+
+      .. code-block:: console
+
+         CONFIG_SB_SIGNING_KEY_FILE="../../_keys/priv.pem"
+
+      Or
+
+      .. code-block:: console
+
+         CONFIG_SB_SIGNING_KEY_FILE="/home/user/ncs/_keys/priv.pem"
+
+   .. group-tab:: Command line
+
+      .. code-block:: console
+
+         -DCONFIG_SB_SIGNING_KEY_FILE=\"../../_keys/priv.pem\"
+
+      Or
+
+      .. code-block:: console
+
+         -DCONFIG_SB_SIGNING_KEY_FILE=\"/home/user/ncs/_keys/priv.pem\"
+
+      Or, if you set an environment variable named ``NCS`` to ``/home/user/ncs``,
+
+      .. code-block:: console
+
+         -DCONFIG_SB_SIGNING_KEY_FILE=\"$NCS/_keys/priv.pem\"
+
+On Unix systems, you can also symbolically link the file into the sample directory, even if git will see it as an untracked file.
+
+.. note::
+
+   The public key string must be an absolute path to the location of the public key file.
+   Environment variables (like ``$HOME``,``$PWD``, or ``$USER``) and the ``~`` character on Unix systems are not expanded when setting an absolute path from a ``prj.conf`` file or Kconfig fragment, but are expanded correctly in key file strings on the command line.
+
+You can find specific configuration options for using keys with this bootloader in :file:`nrf/subsys/bootloader/Kconfig`.
+
+See :ref:`ug_fw_update_keys` for information on how to generate custom keys for a project.
+
+Additionally, the |NSIB| supports the following methods for signing images with private keys:
+
+* :ref:`ug_fw_update_keys_python` - it is the default method, using the :option:`CONFIG_SB_SIGNING_PYTHON`.
+* :ref:`ug_fw_update_keys_openssl` - it uses the :option:`CONFIG_SB_SIGNING_OPENSSL`.
+* :ref:`Using a custom command <ug_bootloader_adding_immutable_b0_custom_signing>` - it uses the :option:`CONFIG_SB_SIGNING_CUSTOM`.
+
+Both Python and OpenSSL methods are automatically handled for the user, whereas using custom commands requires more configuration steps.
```

Comment:
```suggestion
Both Python and OpenSSL methods are automatically handled for the user, whereas using custom commands requires more configuration steps.
```

What handles these? 

Don't need to say 'the user' when the doc is specifically addressing the user.

Reply:
Reworded, please see latest changes.